zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDBMetricExporter.java

```diff
@@ -0,0 +1,150 @@
+  private static final String WRITE_METRICS_PREFIX = "rocksdb_writes";
+
+  private static final RocksDBMetric[] WRITE_METRICS = {
+    new RocksDBMetric("rocksdb.is-write-stopped", WRITE_METRICS_PREFIX, WRITE_METRICS_HELP),
+    new RocksDBMetric(
+        "rocksdb.actual-delayed-write-rate", WRITE_METRICS_PREFIX, WRITE_METRICS_HELP),
+    new RocksDBMetric("rocksdb.mem-table-flush-pending", WRITE_METRICS_PREFIX, WRITE_METRICS_HELP),
+    new RocksDBMetric("rocksdb.num-running-flushes", WRITE_METRICS_PREFIX, WRITE_METRICS_HELP),
+    new RocksDBMetric("rocksdb.num-running-compactions", WRITE_METRICS_PREFIX, WRITE_METRICS_HELP),
+  };
+
+  private final String partition;
+  private final ZeebeDb<ColumnFamilyType> database;
+  private final Class<ColumnFamilyType> columnFamilyTypeClass;
+
+  public ZeebeRocksDBMetricExporter(
+      final String partition,
+      final ZeebeDb<ColumnFamilyType> database,
+      final Class<ColumnFamilyType> columnFamilyTypeClass) {
+    this.partition = Objects.requireNonNull(partition);
+    this.database = Objects.requireNonNull(database);
+    this.columnFamilyTypeClass = Objects.requireNonNull(columnFamilyTypeClass);
+  }
+
+  public void exportMetrics() {
+    final long startTime = System.currentTimeMillis();
+    for (final ColumnFamilyType columnFamilyName : columnFamilyTypeClass.getEnumConstants()) {
+
+      exportMetrics(columnFamilyName, MEMORY_METRICS);
+      exportMetrics(columnFamilyName, LIVE_METRICS);
+      exportMetrics(columnFamilyName, SST_METRICS);
+      exportMetrics(columnFamilyName, WRITE_METRICS);
+    }
+
+    final long elapsedTime = System.currentTimeMillis() - startTime;
+    LOG.trace("Exporting RocksDBMetrics took + " + elapsedTime + " ms");
+  }
+
+  private void exportMetrics(
+      final ColumnFamilyType columnFamilyName, final RocksDBMetric[] metrics) {
+    for (final RocksDBMetric metric : metrics) {
+      try {
+        database
+            .getProperty(columnFamilyName, metric.getPropertyName())
+            .map(Double::parseDouble)
+            .ifPresent(value -> metric.exportValue(partition, columnFamilyName, value));
+      } catch (Throwable t) {
+        LOG.debug(t.getMessage(), t);
+      }
+    }
+  }
+
+  private static final class RocksDBMetric {
+
+    private final String propertyName;
+    private final Gauge gauge;
+
+    private RocksDBMetric(String propertyName, String namePrefix, String help) {
+      this.propertyName = Objects.requireNonNull(propertyName);
+
+      gauge =
+          Gauge.build()
+              .namespace(ZEEBE_NAMESPACE)
+              .name(namePrefix + gaugeSuffix())
+              .help(help)
+              .labelNames(PARTITION, COLUMN_FAMILY_NAME)
+              .register();
+    }
+
+    private String gaugeSuffix() {
+      final String suffix =
+          "_" + propertyName.substring(propertyName.indexOf(".") + 1); // cut off "rocksdb." prefix
```

Comment:
do you think this is necessary? Or is this because dot is not allowed?

Reply:
More an aesthetic preference. Otherwise the metrics might e.g. be called:
`rocksdb_memory_rocksdb.cur-size-all-mem-tables`

Felt this was a bit redundant.
Happy to change it, though.

Reply:
No it's fine thanks